File: D1-Morning/C#/D1-S2/W5-ConstructorAndOverload/W5-ConstructorAndOverload/Person.cs
namespace W5_ConstructorAndOverload
{
    class Person
    {
        public string firstName;
        public string middleName;
        public string lastName;

        public Person()
        {

        }

        public Person(string firstName, string lastName)
        {
            this.firstName = firstName;
            this.lastName = lastName;
        }

        public Person(string firstName, string middleName, string lastName)
            : this(firstName, lastName)
        {
            this.middleName = middleName;
        }

        public string GetMyFullName()
        {
            return firstName + " " + middleName + " " + lastName;
        }

        public string GetFullReverseName()
        {
            string fullName = GetMyFullName();
            string reverseName = "";
            for (int index = fullName.Length - 1; index >= 0; index--)
            {
                reverseName += fullName[index];
            }
            return reverseName;
        }
    }
}

File: D1-Morning/D1-S2-PHP/W5-ConstructorAndOverload/Person.php
<?php

class Person
{
    public $firstName;
    public $middleName;
    public $lastName;

    public function __construct($firstName, $middleName, $lastName)
    {
        $this->firstName = $firstName;
        $this->middleName = $middleName;
        $this->lastName = $lastName;
    }

    public function getFullReverseName()
    {
        $fullName = $this->getMyFullName();
        $reverseName = "";
        for ($index = strlen($fullName) - 1; $index >= 0; $index--) {
            $reverseName = $reverseName . $fullName[$index];
        }
        return $reverseName;
    }

    public function getMyFullName()
    {
        return $this->firstName . " " . $this->middleName . " " . $this->lastName;
    }


}